proto/tendermint/abci/types.proto

```diff
@@ -270,6 +284,18 @@ message ResponseApplySnapshotChunk {
   }
 }
 
+message ResponseExtendVote {
+  tendermint.types.VoteExtension vote_extension = 1;
+}
+
+message ResponseVerifyVoteExtension {
+  // XXX
```

Comment:
???

Reply:
Do we anticipate the `data` field to ever be used, or is it just included for consistency with ResponseDeliverTx & ResponseCheckTx?

To be honest, I don't understand what the data or info fields are used for in other responses either. I have preference for leaving these four fields, and then in a later pass of ABCI++ assess the utility of `data` and `info` everywhere.

Reply:
My mistake, I've changed the response fields to `enum Result` in the second PR and not merged back. I will apply it here.

Reply:
I introduced these first for the consistency with the Response*Tx, but I have found that the recent methods have introduced simple enum style result type, and I thought it would be better for our case. The enum has three result types, each called `ACCEPT, REJECT, ABORT`, where `ACCEPT` and `RESULT` act as validation/invalidation for a vote extension, and `ABORT` acts as a malicious vote extension(such as wrong oracle vote data). I think handling on those `ABORT` type could be done inside the state machine itself, without returning any additional information to Tendermint side.

(Please suggest a better name for `ABORT` if you have any)